src/modules/validators/OwnableValidator.ts

```diff
@@ -1,179 +1,190 @@
-//     }
-
-//     if(receipt.success) {
-//       this.owners = this.owners.filter((o: Address) => o !== owner)
-//     }
-//     return receipt
-
-//   }
-
-//   public async addOwner(owner: Address): Promise<UserOpReceipt> {
-//     const calldata = encodeFunctionData({
-//       functionName: "addOwner",
-//       abi: parseAbi(["function addOwner(address owner)"]),
-//       args: [owner]
-//     })
-//     const response = await this.smartAccount.sendTransaction({
-//       to: this.moduleAddress,
-//       data: calldata,
-//       value: 0n
-//     })
-//     const receipt = await response.wait(5)
-//     if(receipt.success) {
-//       this.owners.push(owner)
-//     }
-//     return receipt
-//   }
-
-//   public async getOwners(): Promise<Address[]> {
-//     try {
-//       const owners = (await this.smartAccount.publicClient.readContract({
-//         address: OWNABLE_VALIDATOR,
-//         abi: parseAbi([
-//           "function getOwners(address account) external view returns (address[])"
-//         ]),
-//         functionName: "getOwners",
-//         args: [await this.smartAccount.getAccountAddress()]
-//       })) as Address[]
-
-//       return owners
-//     } catch (err) {
-//       console.error(err)
-//       return []
-//     }
-//   }
-
-//   public getDummySignature(): Hex {
-//     return "0xfe68e79dbbc872d7a22c6f3eb24f8145edf8d8333092c666221db79bf236e25626e9b05d8230aee0baee7e8b15d60b51e408f1e640b1e0c89deb874b17e269d91b199dfd0340577f695639a048a231a64d181f4e2d66acd5c29ecd0a9b25593c291eb4aba21a15fa4fded885e2c44a49887de1cc9adced5088bcac65cf8c1f29c71c" as Hex
-//   }
-
-//   async signUserOpHash(userOpHash: string): Promise<Hex> {
-//     const signature = await this.signer.signMessage({ raw: userOpHash as Hex })
-//     return signature as Hex
-//   }
-// }
+import {
+    type Address,
+    concat,
+    decodeAbiParameters,
+    encodeAbiParameters,
+    encodeFunctionData,
+    encodePacked,
+    getAddress,
+    parseAbi,
+    parseAbiParameters
+} from "viem"
+import type { Hex } from "viem"
+import type { NexusSmartAccount } from "../../account/NexusSmartAccount.js"
+import { OWNABLE_VALIDATOR_ADDRESS, SENTINEL_ADDRESS, UserOperationStruct } from "../../account/index.js"
+import { BaseValidationModule } from "../base/BaseValidationModule.js"
+import type { Module } from "../utils/Types.js"
+
+export class OwnableValidator extends BaseValidationModule {
+    public smartAccount: NexusSmartAccount
+    public owners: Address[]
+    public threshold: number
+
+    private constructor(
+        moduleConfig: Module,
+        smartAccount: NexusSmartAccount
+    ) {
+        if (!moduleConfig.data) {
+            throw new Error("Module data is required")
+        }
+        const moduleData = decodeAbiParameters(
+            parseAbiParameters("uint256 threshold, address[] owners"),
+            moduleConfig.data
+        )
+        super(moduleConfig, smartAccount.getSmartAccountOwner())
```

Comment:
isn't super supposed to accept a signer?

```
private constructor(moduleConfig: Module, signer: SmartAccountSigner) {
    super(moduleConfig, signer)
  }
```

Reply:
We already have access to the signer from the smart account, so nice catch, this is not used and can be removed

Reply:
That's what smartAccount.getSmartAccountOwner() is passing

Reply:
ok I thought by name it returns an address and not signer object.